File: solutions/elixir/lasagna/2/lib/lasagna.ex
defmodule Lasagna do
  def expected_minutes_in_oven() do
    40
  end

  def remaining_minutes_in_oven(number) do
    expected_minutes_in_oven() - number
  end

  def preparation_time_in_minutes(layer) do
    layer * 2
  end

  def total_time_in_minutes(layer, time) do
    preparation_time_in_minutes(layer) + time
  end

  def alarm() do 
    "Ding!"
  end
end

File: solutions/java/lasagna/1/src/main/java/Lasagna.java
public class Lasagna {

    public static final int MINUTES_IN_OVEN = 40;
    
    public int expectedMinutesInOven(){
        return MINUTES_IN_OVEN;
    }

    public int remainingMinutesInOven(int pastMinutes){
        return MINUTES_IN_OVEN - pastMinutes;
    }

    public int preparationTimeInMinutes(int amountOfLayers){
        return 2 * amountOfLayers;
    }

    public int totalTimeInMinutes(int amountOfLayers, int pastMinutes){
        return preparationTimeInMinutes(amountOfLayers) + pastMinutes;
    }
}
